metrics/tenant_registries_test.go

```diff
@@ -608,11 +611,6 @@ func TestFloat64PrecisionStability(t *testing.T) {
 		cardinality   = 20
 	)
 
-	// Randomise the seed but log it in case we need to reproduce the test on failure.
-	seed := time.Now().UnixNano()
-	rand.Seed(seed)
-	t.Log("random generator seed:", seed)
-
```

Comment:
Instead of dropping the seeding, can you create a seeded generator instead (also make sure to update `rand.*` calls below)?

```
	seed := time.Now().UnixNano()
	rnd := rand.New(rand.NewSource(seed))
	t.Log("random generator seed:", seed)
```

Reply:
Also done.